sway/input/seat.c

```diff
@@ -521,14 +538,15 @@ void seat_set_focus_warp(struct sway_seat *seat,
 		wl_list_remove(&seat_con->link);
 		wl_list_insert(&seat->focus_stack, &seat_con->link);
 
-		if (container->type == C_VIEW && !seat_is_input_allowed(
-					seat, container->sway_view->surface)) {
-			wlr_log(L_DEBUG, "Refusing to set focus, input is inhibited");
-			return;
+		if (last_focus) {
+			unfocus(last_focus, seat);
 		}
 
 		if (container->type == C_VIEW) {
-			seat_send_focus(seat, container);
+			seat_send_focus_iterator(container, seat);
+		} else {
+			container_descendants(container, C_VIEW, seat_send_focus_iterator,
+					seat);
```

Comment:
Something that bugs me but I don't generally mention is functions with bad balance. You don't have to split the line once you get to 80 chars, you can split it earlier too.